src/Context/GtmContext.php

```diff
@@ -75,17 +75,46 @@ public function getDataLayerSettingShouldMatch($key, $regex) {
    * @return mixed
    * @throws \Exception
    */
-  protected function getDataLayerValue($key) {
-    $json_arr = $this->getDataLayerJson();
-
-    // Loop through the array and return the data layer value
-    foreach ($json_arr as $json_item) {
-      if (isset($json_item[$key])) {
-        return $json_item[$key];
-      }
+    protected function getDataLayerValue($key) {
+        $json_arr = $this->getDataLayerJson();
+        // Loop through the array and return the data layer value
+        foreach ($json_arr as $json_item) {
+            // Check if the key contains dot.
+            if (strpos($key, '.', 0) !== false) {
```

Comment:
@yashmit-alshaya Lets move this entire block into a new function, so we keep getDataLayerValue() small.
I suggest the name `getDotValue()`

Reply:
Yes created a new function and changed other as per suggestion.